tests/topotests/multicast_pim_uplink_topo1/test_multicast_pim_uplink_topo1.py

```diff
@@ -0,0 +1,3379 @@
+            "'show ip multicast count json'"
+        )
+
+        result = verify_multicast_traffic(tgen, input_traffic)
+        assert result is True, "Testcase {} : Failed Error: {}".format(tc_name, result)
+
+        step("Shut and no shut the source interface from R4")
+
+        intf_r4_i6 = topo["routers"]["r4"]["links"]["i6"]["interface"]
+        shutdown_bringup_interface(tgen, "r4", intf_r4_i6, False)
+        shutdown_bringup_interface(tgen, "r4", intf_r4_i6, True)
+
+        step(
+            "After shut/no shut of source interface verify all the (S,G) "
+            "got re-learn and IIF/OIF pointing any of the links from R2 or "
+            "R3 verify using 'show ip mroute json'"
+        )
+
+        step(
+            "(S,G) OIL on R1 has only respective receiver port and uplink port "
+            " , RP side oil is removed"
+        )
+
+        for data in input_dict_sg:
+            result = verify_mroutes(
+                tgen,
+                data["dut"],
+                data["src_address"],
+                IGMP_JOIN_RANGE_1,
+                data["iif"],
+                data["oil"],
+            )
+            assert result is True, "Testcase {} : Failed Error: {}".format(
+                tc_name, result
+            )
+
+        step("No change seen on (*,G) mroutes")
+
+        for data in input_dict_starg:
+            result = verify_mroutes(
+                tgen,
+                data["dut"],
+                data["src_address"],
+                IGMP_JOIN_RANGE_1,
+                data["iif"],
+                data["oil"],
+            )
+            assert result is True, "Testcase {} : Failed Error: {}".format(
+                tc_name, result
+            )
+
+    step(
+        "Traffic is received for all the groups , verify using "
+        "'show ip multicast count json'"
+    )
+
+    result = verify_multicast_traffic(tgen, input_traffic)
+    assert result is True, "Testcase {} : Failed Error: {}".format(tc_name, result)
+
+    step(
+        "Shut source interface from R4 and no shut immediate after the "
+        "same source upstream expires from DUT"
+    )
+
+    intf_r4_i6 = topo["routers"]["r4"]["links"]["i6"]["interface"]
+    shutdown_bringup_interface(tgen, "r4", intf_r4_i6, False)
+
+    step(
+        "After shut verify upstream got expired using 'show ip pim "
+        "'upstream' before doing no shut"
+    )
+
+    done_flag = False
+    for retry in range(1, 11):
+        result = verify_upstream_iif(
+            tgen, "r1", "Unknown", source_i6, IGMP_JOIN_RANGE_1, expected=False
+        )
+        if result is not True:
+            done_flag = True
+        else:
+            continue
+
+        if done_flag:
+            logger.info("Expected Behavior: {}".format(result))
+            break
+
+    assert done_flag is True, (
+        "Testcase {} : Failed Error: \n "
+        "mroutes are still present, after waiting for 10 mins".format(tc_name)
```

Comment:
Do we need to wait 10 minutes to make sure it failed? Or is this a misleading assert message?

Reply:
Hi @ton31337 , There was a step to wait for 10 mins for traffic to get clear. But to remove 10 mins wait, we have enhanced the script, now we are stopping hosts itself so traffic is getting cleared faster. I have updated assert message. Thank you.